Mathlib/SetTheory/Cardinal/Finite.lean

```diff
@@ -337,14 +338,42 @@ theorem card_le_one_iff_subsingleton (α : Type*) : card α ≤ 1 ↔ Subsinglet
   rw [← le_one_iff_subsingleton]
   simp [card]
 
+lemma card_eq_one_iff_unique {α : Type*} : card α = 1 ↔ Nonempty (Unique α) := by
+  rw [unique_iff_subsingleton_and_nonempty α, le_antisymm_iff, one_le_iff_ne_zero]
+  exact and_congr (card_le_one_iff_subsingleton α) (card_ne_zero_iff_nonempty α)
+
 theorem one_lt_card_iff_nontrivial (α : Type*) : 1 < card α ↔ Nontrivial α := by
   rw [← Cardinal.one_lt_iff_nontrivial]
   conv_rhs => rw [← Nat.cast_one]
   rw [← natCast_lt_toENat_iff]
   simp only [ENat.card, Nat.cast_one]
 
 @[simp]
-theorem card_prod (α β : Type*) : ENat.card (α × β) = .card α * .card β := by
+theorem card_prod (α β : Type*) : ENat.card (α × β) = card α * card β := by
   simp [ENat.card]
 
+@[simp]
+lemma card_pow {α β : Type*} : ENat.card (α → β) = (card β) ^ card α := by
+  classical
+  rcases isEmpty_or_nonempty α with α_emp | α_emp
+  · simp [(card_eq_zero_iff_empty α).2 α_emp]
+  rcases finite_or_infinite α
+  · rcases finite_or_infinite β
+    · have _ := Fintype.ofFinite α
+      have _ := Fintype.ofFinite β
```

Comment:
```suggestion
    · letI := Fintype.ofFinite α
      letI := Fintype.ofFinite β
```
Fintype is data, so it's good practice to introduce it with let (even if it's not strictly necessary here).